dask_expr/_collection.py

```diff
@@ -1791,6 +1851,25 @@ def nbytes(self):
     def keys(self):
         return self.index
 
+    def __array_wrap__(self, array, context=None):
+        if isinstance(context, tuple) and len(context) > 0:
+            if isinstance(context[1][0], np.ndarray) and context[1][0].shape == ():
+                index = None
+            else:
+                index = context[1][0].index
+        else:
+            try:
+                import inspect
+
+                method_name = f"`{inspect.stack()[3][3]}`"
+            except IndexError:
+                method_name = "This method"
+            raise NotImplementedError(
+                f"{method_name} is not implemented for `dask.dataframe.Series`."
+            )
+
+        return meta_series_constructor(self)(array, index=index, name=self.name)
```

Comment:
Hmm, two very similar `__array_wrap__` methods. What do you think about moving this whole thing into `FrameBase` dispatching to a `meta_constructor` which routes to the appropriate `meta_series/frame_constructor`? Or maybe a class attr in Series/DataFrame for the correct constructor, dependency injection style.

Reply:
Yeah makes sense as a follow up although I am not sure if I am missing something, I took this mostly as is from dask/dask